addons/bus/static/src/services/bus_service.js

```diff
@@ -73,6 +79,20 @@ export const busService = {
         function handleMessage(messageEv) {
             const { type, data } = messageEv.data;
             switch (type) {
+                case "provide_logs": {
+                    const blob = new Blob([JSON.stringify(data, null, 2)], {
+                        type: "application/json",
+                    });
+                    const url = URL.createObjectURL(blob);
+                    const a = document.createElement("a");
+                    a.href = url;
+                    a.download = `bus_logs_${luxon.DateTime.now().toFormat(
+                        "yyyy-LL-dd-HH-mm-ss"
+                    )}.json`;
+                    a.click();
+                    URL.revokeObjectURL(url);
```

Comment:
Can we use `window.location = url` instead of a fake click?

Reply:
Redirecting the current page is a bit harsh. Downloading the file is better IMO.